src/Compilation.zig

```diff
@@ -451,6 +451,9 @@ pub const CacheMode = link.CacheMode;
 pub const LinkObject = struct {
     path: []const u8,
     must_link: bool = false,
+    // When linking an object via -l :path/to/libX.so, this path should be put
+    // to the binary instead of `path`.
+    shlib_path: ?[]const u8 = null,
```

Comment:
Hmm, this looks sus to me. Is it possible for `LinkObject` to have `path` AND `shlib_path` set at the same time? If so, what does that even mean? If not, I think it would make more sense to convert `shlib_path` into some kind of a flag like `is_shlib: bool = false`. From your changes, I believe the latter is really what we want.

Reply:
This was my first attempt until I found

https://github.com/ziglang/zig/blob/9d179a98f69dbab393cbb3fc5dd4b64c553a721b/src/main.zig#L2659

Until your comment I *thought* they have to be together, but now I am not so sure. I will do more testing.